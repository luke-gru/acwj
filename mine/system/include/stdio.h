#ifndef _STDIO_H_
#define _STDIO_H_

#include <stddef.h>
#include <stdarg.h>

#define EOF (-1)

// This FILE definition will do for now
typedef char *FILE;
extern FILE *stderr;
extern FILE *stdout;

FILE *fopen(char *pathname, char *mode);
size_t fread(void *ptr, size_t size, size_t nmemb, FILE *stream);
size_t fwrite(void *ptr, size_t size, size_t nmemb, FILE *stream);
int fclose(FILE *stream);
int printf(char *format);
int fprintf(FILE *stream, char *format);
size_t getline(char **lineptr, size_t *n, FILE *stream);

FILE *popen(const char *command, const char *type);
int pclose(FILE *stream);

int vfprintf(FILE *stream, const char *format, va_list ap);

#endif	// _STDIO_H_
