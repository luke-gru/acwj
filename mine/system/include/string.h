#ifndef _STRING_H
#define _STRING_H_
#endif
