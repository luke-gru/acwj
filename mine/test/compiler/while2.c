#include <stdio.h>
#include <string.h>

char *string = "\t\f\t hello and goodbye for now";
int pos = 0;

int next(void) {
  int res;
  if (pos == strlen(string)) return (EOF);
  res = string[pos];
  pos++;
  return (res);
}

static int skip(void) {
  int c;

  c = next();
  while (' ' == c || '\t' == c || '\n' == c || '\r' == c || '\f' == c) {
    if (c == EOF) return (EOF);
    c = next();
  }
  return (c);
}

int main() {
  int c;
  while ((c = skip()) != EOF) {
    printf("c: %c, pos: %d\n", c, pos);
  }
  return (0);
}

