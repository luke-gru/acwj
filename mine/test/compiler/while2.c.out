c: h, pos: 5
c: e, pos: 6
c: l, pos: 7
c: l, pos: 8
c: o, pos: 9
c: a, pos: 11
c: n, pos: 12
c: d, pos: 13
c: g, pos: 15
c: o, pos: 16
c: o, pos: 17
c: d, pos: 18
c: b, pos: 19
c: y, pos: 20
c: e, pos: 21
c: f, pos: 23
c: o, pos: 24
c: r, pos: 25
c: n, pos: 27
c: o, pos: 28
c: w, pos: 29
